### PR TITLE
Beef up compartment search tests

### PR DESCRIPTION
**Pull request by lmsurpre**
In prep for exploring some ideas around supporting logical 'or' on the
compartment search param.

Signed-off-by: Lee Surprenant <lmsurpre@us.ibm.com>

### CODE REVIEW

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java [hunk 1]
```diff
@@ -1,61 +1,198 @@
...
+        assertCompartmentSearchReturnsSavedResource(PRACTITIONER, PRACTITIONER_ID, "uri", "urn:uuid:53fefa32-1111-2222-3333-55ee120877b7");
+        assertCompartmentSearchDoesntReturnSavedResource(PATIENT, OTHER_ID, "uri", "urn:uuid:53fefa32-1111-2222-3333-55ee120877b7");
+        assertCompartmentSearchDoesntReturnSavedResource(DEVICE, PATIENT_ID, "uri", "urn:uuid:53fefa32-1111-2222-3333-55ee120877b7");
+    }
+
+    @Test
+    public void testSearchPatientCompartment_relativeReference_chained_number() throws Exception {
+        assertCompartmentSearchReturnsComposition(PATIENT, PATIENT_ID, "subject:Basic.integer", "12");
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.integer", "12");
+        assertCompartmentSearchDoesntReturnSavedResource(PATIENT, OTHER_ID, "subject:Basic.integer", "12");
```
[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java:117]: yes, good catch

[the remark this answers is not in the document]
Comment (lmsurpre) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java]: fixed

**Review by michaelwschroeder — APPROVED** [review]
LGTM

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java [hunk 2]
```diff
@@ -1,61 +1,198 @@
...
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.string", "testString");
+        assertCompartmentSearchDoesntReturnComposition(PATIENT, OTHER_ID, "subject:Basic.string", "testString");
+        assertCompartmentSearchDoesntReturnComposition(DEVICE, PATIENT_ID, "subject:Basic.string", "testString");
+    }
+
+    @Test
+    public void testSearchPatientCompartment_relativeReference_chained_token() throws Exception {
+        assertCompartmentSearchReturnsComposition(PATIENT, PATIENT_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
+        assertCompartmentSearchReturnsComposition(PRACTITIONER, PRACTITIONER_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
+        assertCompartmentSearchDoesntReturnSavedResource(PATIENT, OTHER_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
```
Comment (tbieste) [inline fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java:150]: Should this be assertCompartmentSearchDoesntReturnComposition?

**Comment by tbieste** [comment]
Left one more comment, otherwise LGTM.

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java (hunk 2 above)
Comment (lmsurpre) [reply fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java:150]: drat, I thought I got all the places...definitely missed this line (and the following one)

File: fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java [hunk 3]
```diff
@@ -1,61 +1,198 @@
...
+        assertCompartmentSearchDoesntReturnSavedResource(DEVICE, PATIENT_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
```
Comment (lmsurpre) [inline fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchCompartmentTest.java:151]:
```suggestion
        assertCompartmentSearchDoesntReturnComposition(PATIENT, OTHER_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
        assertCompartmentSearchDoesntReturnComposition(DEVICE, PATIENT_ID, "subject:Basic.CodeableConcept", "http://example.org/codesystem|code");
```

**Review by tbieste — APPROVED** [review]

**Pull request merged**